doc/migration.md

```diff
@@ -0,0 +1,415 @@
+std::optional<sf::IntRect> intersection = rect1.findIntersection(rect2);
+// position={100, 100} size={100, 100}
+```
+
+## `sf::Angle`
+
+All angles are now represented with a strong type named `sf::Angle`.
+This type provides two functions for creating angles called `sf::degrees(float)` and `sf::radians(float)` which construct an angle from either some value of degrees or radians.
+Operators (`+`, `-`, etc.) are provided to perform mathematical operations with angles.
+If you need access to the raw angle as a `float` then you can use either `sf::Angle::asDegrees()` or `sf::Angle::asRadians()`.
+
+v2:
+```cpp
+sf::RectangleShape shape(sf::Vector2u(50, 50));
+shape.setRotation(90);
+std::cout << "Rotation: " << shape.getRotation() << '\n';
+```
+
+v3:
+```cpp
+sf::RectangleShape shape({50, 50});
+shape.setRotation(sf::degrees(90));
+std::cout << "Rotation: " << shape.getRotation().asDegrees() << '\n';
+```
+
+## Renamed Functions
+
+A number of functions have new names but otherwise have not changed their semantics.
+
+| v2                                | v3                                |
+| --------------------------------- | --------------------------------- |
+| `sf::Font::loadFromFile`          | `sf::Font::openFromFile`          |
+| `sf::Socket::getHandle`           | `sf::Socket::getNativeHandle`     |
+| `sf::WindowBase::getSystemHandle` | `sf::WindowBase::getNativeHandle` |
+| `sf::Texture::create`             | `sf::Texture::resize`             |
+| `sf::RenderTexture::create`       | `sf::RenderTexture::resize`       |
+| `sf::Image::create`               | `sf::Image::resize`               |
+| `sf::Sound::getLoop`              | `sf::Sound::isLooping`            |
+| `sf::Sound::setLoop`              | `sf::Sound::setLooping`           |
+| `sf::SoundStream::getLoop`        | `sf::SoundStream::isLooping`      |
+| `sf::SoundStream::setLoop`        | `sf::SoundStream::setLooping`     |
+
+## Removal of Default Constructors
+
+The default constructors `sf::Sound::Sound()`, `sf::Text::Text()`, and `sf::Sprite::Sprite()` were removed.
+They can be replaced by the corresponding constructors which accept a resource type.
+
+| v2                     | v3                                         |
+| ---------------------- | ------------------------------------------ |
+| `sf::Sound::Sound()`   | `sf::Sound::Sound(const sf::SoundBuffer&)` |
+| `sf::Text::Text()`     | `sf::Text::Text(const sf::Font&)`          |
+| `sf::Sprite::Sprite()` | `sf::Sprite::Sprite(const sf::Texture&)`   |
+
+Now that these classes are guaranteed to be holding a reference to their corresponding resource type, the functions used to access to those resources can return a reference instead of a pointer.
+These functions are `sf::Sound::getBuffer()`, `sf::Text::getFont()`, and `sf::Sprite::getTexture()`.
+
+## `std::optional` Usage
+
+SFML 3 makes liberal use of `std::optional` to express when a given function may not return a value.
```

Comment:
If you find a good explainer you like I'm fine adding it in here. However I'm inclined to believe users won't struggle to understand `std::optional` since it's already so similar to `std::unique_ptr`. It's really not that complicated.